lib/osut/utils.rb

```diff
@@ -4391,76 +4548,87 @@ def genAnchors(s = nil, set = [], tag = :vtx)
       end
     end
 
-    if facingUp?(pts)
-      if xyz?(pts, :z)
-        dZ = 0
+    set.each_with_index do |st, i|
+      # When a subset already holds a leader line anchor (from an initial call
+      # to 'genAnchors'), it inherits key :out - a Hash holding (among others) a
+      # 'realigned' set of points (by default a 'realigned' :box). The latter is
+      # typically generated from an outdoor-facing roof (e.g. when called from
+      # 'lights'). Subsequent calls to 'genAnchors' may send (as first
+      # argument) a corresponding ceiling tile below (also from 'addSkylights').
+      # Roof vs ceiling may neither share alignment transformation nor space
+      # site transformation identities. All subsequent calls to 'genAnchors'
+      # shall recover the :out points, apply a succession of de/alignments and
+      # transformations in sync , and overwrite tagged points.
```

Comment:
This redesign (or rather a more systematic application of an original principle) became necessary when trying to manage multiple skylight arrays for a single roof surface (e.g. a large unconditioned attic roof sheltering a dozen conditioned spaces below).